src/lib/connection/Subscription.ts

```diff
@@ -83,6 +45,27 @@ export class Subscription {
         this.behaviourBind.push(behaviour);
     }
 
-    private getTopic = () => this.mqttSensorQueue.getTopic(this.type, this.header);
+    public onMessage(message: object) {
+        this.mqttSensorQueue.callback(this.getTopic(), message);
+    }
+
+    public getTopic = () => this.mqttSensorQueue.getTopic(this.type, this.header);
+}
+
+export class ContextSubscription extends Subscription {
```

Comment:
Ich find die Klassen kannst du einsparen und in der Connection einfach normale Subscription Objekte zurückgeben.
Wird dadurch nicht komplizierter.
Also zB Connection.subscribeContext

`public subscribeContext(header: QueueHeader, callback: QueueCallback) {
        const subscription = new Subscription(ConversationQueueType.CONTEXT, header, callback, this.mqttSensorQueue);
        this.setSubscription(subscription);
        return subscription;
    }`

Reply:
habs auf normale Subscription umgeändert 👍